metalbear-co/mirrord
# Add E2E tests

This PR closes #35 adding tests conforming to the suite provided in #50 

Current state of the tests:
validation tests:
- run the node application with mirrord exec in a forked process
- send GET, POST, PUT, and DELETE requests and keep a counter
- POST validates data.txt, PUT creates a file with the data received, DELETE reads the file created by PUT, validates the data and removes the file. 
- if all operations succeed, we exit with EXIT_SUCCESS
- since only a single job and pod is created, we verify that these resources exist and are removed after agent timeout finishes. 

namespace tests [agent]:
- create a new namespace 
- run a binary with agent namespace assigned the new namespace
- check if the job and the pod exist in the created namespace within the timeout limit 
- run a binary with agent namespace assigned a non-existent namespace 
- make sure the process crashes
- verify no resources have been created in either the default or the bad namespace 

namespace tests [pod]:
- create a new namespace 
- run a binary with pod namespace assigned the new namespace in the form of an env var
- verify the job exists
